### PR TITLE
JSON Metrics integration

### PR DESCRIPTION
**Pull request by UDXS**
This PR is part of work to replace the grep-based JSON metrics extraction in OR and ORFS with JSON logging that is integrated into the App.
Currently, it lacks the `__clock__`  set of metrics.

`genMetrics.py` is now only used to extract runtime information (build, time, resource usage) and to merge metrics between the different flow stages. The original `genMetrics.py` is available as `genMetricsGrep.py` as a temporary fallback should it be needed.

### CODE REVIEW

File: flow/scripts/report_metrics.tcl [hunk 1]
```diff
@@ -76,3 +82,41 @@ proc report_metrics { when {include_erc true} } {
 
   puts ""
 }
+
+# Ex. report_class = "timing__setup__tns"
+proc map_when_to_json_key {when report_class} {
```
Comment (maliberty) [inline flow/scripts/report_metrics.tcl:166]: Where is this used?

File: flow/scripts/write_ref_sdc.tcl [hunk 2]
```diff
@@ -18,6 +18,8 @@ if { [llength $clks] == 0 } {
     set ref_period [expr ($period - $slack) * (1.0 - $margin/100.0)]
     utl::info "FLW" 8 "Clock $clk_name period [format %.3f $ref_period]"
     utl::info "FLW" 9 "Clock $clk_name slack [format %.3f $slack]"
+    utl::metric_float "timing__clock__slack" $slack
+    utl::metric_float "timing__clock__period" $slack
```
Comment (maliberty) [inline flow/scripts/write_ref_sdc.tcl]: $slack -> $period

**Review by maliberty — COMMENTED** [review]
Mostly fine but I'll hold off until its brother PR is ready and they can be tested together.

**Comment by vvbandeira** [comment]
Replaced by #519

**Pull request closed without merging**